transmart-core-db/grails-app/services/org/transmartproject/db/clinical/MultidimensionalDataResourceService.groovy

```diff
@@ -708,6 +767,25 @@ class MultidimensionalDataResourceService implements MultiDimensionalDataResourc
         queryResult
     }
 
+    @Override QueryResult findQueryResultByConstraint(String constraintText, User user) {
+        QtQueryMaster queryMaster = getQueryMasterByConstraint(user, constraintText)
+        if (queryMaster == null) {
+            log.info "Patient set not found with specified constraints."
+            return null
+        }
+        QtQueryInstance queryInstance = queryMaster.queryInstances.find {
+            it.startDate == queryMaster.queryInstances*.startDate.max() && it.deleteFlag == 'N'
```

Comment:
This looks like sorting.

Reply:
what if a row with `deleteFlag == 'Y'` would have a maximum start date?